src/components/button/button.component.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../icon';
+import StyledButton, { StyledButtonSubtext } from './button.style';
+import tagComponent from '../../utils/helpers/tags';
+import Link from '../link';
+import OptionsHelper from '../../utils/helpers/options-helper';
+
+const Button = (props) => {
+  const {
+    as,
+    disabled,
+    iconPosition,
+    iconType,
+    theme,
+    ...rest
+  } = props;
+
+  if (props.href || props.to) {
+    return (
+      <Link
+        role='button'
+        variant={ theme }
+        { ...tagComponent('button', props) }
+        { ...rest }
+      >
+        { renderChildren(props) }
+      </Link>
```

Comment:
it could be extracted to a separate method, e.g.: `renderLink` to increase readability

Reply:
Yeah I agree it's a bit convoluted, I'll make the suggested change